joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/types/DefaultTypeInfoProvider.scala

```diff
@@ -504,36 +430,34 @@ class DefaultTypeInfoProvider(environment: KotlinCoreEnvironment) extends TypeIn
       case Some(fnDescriptor) =>
         val originalDesc          = fnDescriptor.getOriginal
         val renderedFqNameForDesc = TypeRenderer.renderFqName(originalDesc)
-        val renderedFqName =
+
+        val nameExtensionReceiverMaybe =
           if (originalDesc.getExtensionReceiverParameter != null) {
-            val extType = originalDesc.getExtensionReceiverParameter.getType
-            val extName = originalDesc.getName
-            val rendered =
-              if (renderedFqNameForDesc.startsWith(TypeConstants.kotlinApplyPrefix)) {
-                TypeConstants.javaLangObject
-              } else {
-                TypeRenderer.render(extType, false, false)
-              }
-            s"$rendered.$extName"
+            Some((originalDesc.getName, originalDesc.getExtensionReceiverParameter))
```

Comment:
The change to this block in general makes it SOOOOO much cleaner than before, but the new version can be rewritten as 
```
Option(originalDesc.getExtensionReceiverParameter).map { extensionReceiverParameter =>
  (originalDesc.getName, extensionReceiverParameter)
}
```
which is, in my opinion, nicer than the explicit null check (I really hate null ;))

Reply:
This comment should actually include the 
```
 } else {
  None
  }
  ```
  below

Reply:
i'll leave this one as it is for now